langchain/text_splitter.py

```diff
@@ -90,12 +95,16 @@ def _merge_splits(self, splits: Iterable[str], separator: str) -> List[str]:
                     # - we have a larger chunk than in the chunk overlap
                     # - or if we still have any chunks and the length is long
                     while total > self._chunk_overlap or (
-                        total + _len > self._chunk_size and total > 0
+                        total + _len + (separator_len if len(current_doc) > 0 else 0)
```

Comment:
@hwchase17 Sorry, isn't ` total + _len + (separator_len if len(current_doc) > 0 else 0) > self._chunk_size` always true at this point? 

And if `self._chunk_overlap` is expected to always be greater than `0`, this condition reduces to  `total > 0`?